geode-core/src/test/java/org/apache/geode/test/dunit/rules/GfshShellConnectionRule.java

```diff
@@ -173,6 +173,27 @@ public void connect(int port, PortType type, String... options) throws Exception
     connected = (result.getStatus() == Result.Status.OK);
   }
 
+  // when calling this method, the test needs to be in either geode-web or geode-assembly
+  public void executeOverJmxAndHttp(Member member, Runnable runnable) throws Exception {
+    connectAndVerify(member.getJmxPort(), PortType.jmxManger);
+    runnable.run();
+    disconnect();
+    connectAndVerify(member.getHttpPort(), PortType.http);
+    runnable.run();
+    disconnect();
+  }
+
+  // when calling this method, the test needs to be in either geode-web or geode-assembly
+  public void secureExecuteOverJmxAndHttp(Member member, String username, String password,
```

Comment:
These are excellent additions, but are currently unused.  `ExecuteFunctionCommandsWithSecurityDUnitTest` looks like a good candidate for application.